e2e/test/scenarios/visualizations/object_detail.cy.spec.js

```diff
@@ -269,3 +273,60 @@ function changeSorting(columnName, direction) {
   });
   cy.wait("@dataset");
 }
+
+['postgres', 'mysql'].forEach(dialect => {
+  describe(`Object Detail > composite keys (${dialect})`, { tags: ['@external'] }, () => {
+    const TEST_TABLE = "composite_pk_table";
+
+    beforeEach(() => {
+      resetTestTable({ type: dialect, table: TEST_TABLE });
+      restore(`${dialect}-writable`);
+      cy.signInAsAdmin();
+      resyncDatabase(WRITABLE_DB_ID);
+      cy.wait(300)
```

Comment:
Is there a way to reliably wait for the resync? For instance, pooling the sync state from the server. I am worried that at some point this can bring flakes

Reply:
👍  I'll take another look at the `waitForSyncToFinish` function, I think I can maybe make it more robust

Reply:
I found a much better way to check the sync without a wait. thanks for pushing on this!